### DISCUSSION TITLE
Does find queries fetch all the data from Postgres for a collection and then do filtering in-memory?

### DISCUSSION THREAD

**Discussion by shekharx** [FerretDB/FerretDB#903] [General]
I could be wrong but looking at the `msg_find.go` I get the understanding that we fetch all the documents from Postgres and then we do filtering in FerretDB process memory. Is this right? May be I am missing something here. 

```go
fetchedDocs, err := h.fetch(ctx, sp)
for _, doc := range fetchedDocs {
     matches, err := common.FilterDocument(doc, filter)
     resDocs = append(resDocs, doc)
}
```

**Comment by AlekSi** [comment]
That's correct. See "What's Changed" section in [that release](https://github.com/FerretDB/FerretDB/releases/tag/v0.1.0). We will publish a blog post soon explaining that in more detail and how we will change that before 1.0.
